src/frame.js

```diff
@@ -0,0 +1,31 @@
+export default class Frame {
+  constructor() {
+    this.rolls = [];
+    this.pinsLeft = 10;
+    this.result = "";
+    this.score = 0;
+  }
+
+  roll(pins, bonus=false) {
+    if (this.pinsLeft < pins) {throw "Not enough pins left"}
+    this.rolls.push(pins);
+    this.score += pins;
+    if (bonus===false) {
+      this.pinsLeft -= pins;
+    } 
+    if (this.pinsLeft===0 && this.rolls.length===1) {
+      this.result = "Strike";
+    } else if (this.pinsLeft===0 && this.rolls.length===2) {
+      this.result = "Spare";
+    }
```

Comment:
Instead of calculating the result and setting it on the class, a better approach would be to make `result()` and `score()`  (and maybe `pinsleft`) methods that calculates on the fly based on the rolls. This avoids duplication of information (here basically the info is stored 4 times, in different guises), and means this roll method is only responsible for updating the rolls.

eg:
```
score(){
  this.rolls.reduce(function(a, b){ return a + b })
}

result() {
  if (this.rolls.length > 0 && this.rolls[0] === 10) {
    return "Strike"
  } 
  // .... etc
}
```

Reply:
You can also probably get rid of pinsLeft that way. Less duplication of information is better, because it avoids mistakes and inconsistency between sources of truth.